comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +24,225 @@ public static bool IsDirectory(string path)
 		// Return the total size, in bytes, of all the files below the given directory
 		public static long GetTotalSize(string directory)
 		{
-			throw new NotImplementedException();
+            long totalSize = 0;
+            Stack<string> subDirectories = new Stack<string>();
+
+            if (IsDirectory(directory))
+            {
+                subDirectories.Push(directory);
+
+                while (subDirectories.Count > 0)
+                {
+                    string currentDirectory = subDirectories.Pop();
+                    string[] newDirectories = Directory.GetDirectories(currentDirectory);
+
+                    for (int i = 0; i < newDirectories.Length; i++)
+                    {
+                        subDirectories.Push(newDirectories[i]);
+                    }
```

Comment:
This depth-first search code appears in basically all of your functions -- could it have been factored out into a helper function?